src/app/utils/field-validation/field-validation.guards.ts

```diff
@@ -61,9 +61,7 @@ export const isProcessedAttachmentResponse = (
 ): response is ProcessedAttachmentResponse => {
   return (
     response.fieldType === BasicField.Attachment &&
-    'filename' in response &&
-    typeof response.filename === 'string' &&
     'answer' in response &&
-    typeof response.answer === 'string'
+    typeof response.answer === 'string' // No check for response.filename as response.filename is generated only when actual file is uploaded
```

Comment:
could I get your help to clarify this comment? the filename isn't generated before the validators?

Reply:
It is; however the typeguards are there to identify if the shape of the response is valid before the validators are run. Hidden fields - which still return an empty response - will not have response.filename property. Hence checking for filename as a compulsory property for attachment response will result in 'Invalid response shape' error for hidden fields, which is not correct. I've added more details to the comment for clarity